### DISCUSSION TITLE
issue reseting values to undefine, display in React old Value

### DISCUSSION THREAD

**Discussion by carlosLMZarego** [pmndrs/valtio#860] [Bug report]
hi, good day, I've an state set like this

```ts
const user = proxy({
 data :{
  name : "tom",
  color : "red"
 }})
```

suppose that color is optional, so could be undefined, if I reset my data like this

```ts
user.data = {
 name : ""
}
```

in react I'm using useSnapshot

```ts
const user = useSnapshot(scheduleSelected).data
```

in react name is correctly clear but age still shows "red" as value, although the value must set to undefined

same problem happens when I set this to  undefine manually

```
user.data = {
 name : "",
color : undefined
}

what could be the best way to solve this?

**Comment by dai-shi** [comment]
https://livecodes.io/?x=id/b7musf4nkg6
I just created an example to reproduce your case, but it seems to work fine.
Can you create a reproduction?

**Comment by carlosLMZarego** [comment]
hi @dai-shi your code is really similar to mine (mine is a real project with several states but I'd need to find how reproduce this, I'd need to check this more deeply in the weekend because I know that the clear function is being called but when I put

```
user.data = {
  name : undefined,
  age : undefined 
}
```

doesn't work...but when I call this as

```
user.data.name = undefined
user.data.age = undefined
```

it works as expected, it's weird, I'd need check what is happening, right now I noticed that in the devtools the value is updated correctly, so seems a problem with the react component, but I'm using useSnapshot correctly, my project is using NextJS, it doesn't seem to be the problem because as I said, the function prints a console.log indicating that the state is reset, but my component doesn't display the change...thanks for your reply, I'll come back when I can reproduce this error

**Reply by dai-shi** [reply]
Maybe something like this might help:
```diff
- const user = useSnapshot(scheduleSelected).data
+ const { data: user } = useSnapshot(scheduleSelected)
```
